<a href="https://colab.research.google.com/github/HusainHikmal/PRTKM-ML/blob/main/UTS1_G_211_21_0065.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
#Import Library
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from sklearn.linear_model import LinearRegression
import seaborn as sns
%matplotlib inline

In [ ]:
#Membaca Dataset
df = pd.read_csv('CarPrice_Assignment.csv')
df.head()

car_ID  symboling                   CarName fueltype aspiration doornumber  \
0       1          3        alfa-romero giulia      gas        std        two   
1       2          3       alfa-romero stelvio      gas        std        two   
2       3          1  alfa-romero Quadrifoglio      gas        std        two   
3       4          2               audi 100 ls      gas        std       four   
4       5          2                audi 100ls      gas        std       four   

       carbody drivewheel enginelocation  wheelbase  ...  enginesize  \
0  convertible        rwd          front       88.6  ...         130   
1  convertible        rwd          front       88.6  ...         130   
2    hatchback        rwd          front       94.5  ...         152   
3        sedan        fwd          front       99.8  ...         109   
4        sedan        4wd          front       99.4  ...         136   

   fuelsystem  boreratio  stroke compressionratio horsepower  peakrpm citympg  \
0        mpfi       3.47    2.68              9.0        111     5000      21   
1        mpfi       3.47    2.68              9.0        111     5000      21   
2        mpfi       2.68    3.47              9.0        154     5000      19   
3        mpfi       3.19    3.40             10.0        102     5500      24   
4        mpfi       3.19    3.40              8.0        115     5500      18   

   highwaympg    price  
0          27  13495.0  
1          27  16500.0  
2          26  16500.0  
3          30  13950.0  
4          22  17450.0  

[5 rows x 26 columns]

In [ ]:
df.shape

(205, 26)

In [ ]:
# LINIER REGRESI
#Pada regresi linier disini menggunakan variabel/Independent variabel [‘horsepower’] untuk memprediksi harga mobil
X = df[['horsepower']]
Y = df['price']
# MULTIPLE LINIER REGRESI sedangkan pada multiple regresi menggunakan variabel/Independent variabel ‘horsepower’, ‘curbweight’, ‘enginesize’, ‘highwaympg’.
Z = df[['highwaympg', 'curbweight', 'enginesize', 'horsepower']]

Modelling

In [ ]:
# LINIER REGRESI
lm = LinearRegression()
lm.fit(X,Y)
Yhat=lm.predict(X)
Yhat[0:5]

array([14400.43827331, 14400.43827331, 21420.749895  , 12931.07072458,
       15053.49051719])

In [ ]:
lm.intercept_

-3721.7614943227636

In [ ]:
lm.coef_

array([163.26306097])

linier Regresi
price = -3721.76 + 163.26 x horsepower

In [ ]:
# MULTIPLE LINIER REGRESI
lm.fit(Z, df['price'])
LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None)

LinearRegression()

In [ ]:
lm.intercept_

-15762.282866748192

In [ ]:
lm.coef_

array([40.71172629,  4.63040004, 82.68402196, 52.446332  ])

Multiple Linier Regresi
Price = -15762.282866748148 + 40.71172629 x horsepower + 4.63040004 x curbweight + 82.68402196 x enginesize + 52.446332 x horsepower


Evaluasi


In [ ]:
# LINIER REGRESI
lm.fit(X, Y)
print('The R-square is: ', lm.score(X, Y))

The R-square is:  0.653088356490231


In [ ]:
# MULTIPLE LINIER REGRESI
lm.fit(Z, df['price'])
print('The R-square is: ', lm.score(Z, df['price']))

The R-square is:  0.8141577183900568


multiple linier regresi memberikan hasil predikasi yang lebih baik dibandingkan dengan linier regresi.